frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -143,10 +148,11 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
       transformLabel,
       allSelectorItem,
       showBadge = false,
+      appendItems,
     }: ResourceDropdownProps,
     updateSelection: boolean,
   ) => {
-    const unsortedList = {};
+    let unsortedList = {};
```

Comment:
Can you not simply init the list upfront?
```suggestion
    const unsortedList = { ...appendItems };
```